packages/router/router.ts

```diff
@@ -741,6 +743,8 @@ export function createRouter(init: RouterInit): Router {
     "You must provide a non-empty routes array to createRouter"
   );
 
+  let decodeResponse = init.decodeResponse;
```

Comment:
If we get rid of the `defaultDecode` param, I'd then just lift the defaulting logic up to here and then we don't need to do the option detection each time in `callLoaderOrAction`:

```
let decodeResponse = init.decodeResponse || defaultDecodeResponse;
```

(We'd put `defaultDecodeResponse` up next to `defaultMapRouteProperties`)